clients/client-core/src/client/real_messages_control/real_traffic_stream.rs

```diff
@@ -68,6 +70,90 @@ impl Config {
     }
 }
 
+struct SendingDelayController {
+    /// Multiply the average sending delay.
+    /// This is normally set to unity, but if we detect backpressure we increase this
+    /// multiplier. We use discrete steps.
+    current_multiplier: u32,
+
+    /// Maximum delay multiplier
+    upper_bound: u32,
+
+    /// Minimum delay multiplier
+    lower_bound: u32,
+
+    /// To make sure we don't change the multiplier to fast, we limit a change to some duration
+    time_when_changed: Instant,
```

Comment:
I know it's just a draft, but wasm client will blow up here at runtime. `wasm32-unknown-unknown` target doesn't play nice with any tokio-based `time` types

Reply:
oof, ok that's good to know!